sympy/vector/operators.py

```diff
@@ -0,0 +1,187 @@
+
+    def doit(self):
+        return Del(self._get_coord_sys_from_expr(self._scalar)).gradient(self._scalar).doit()
+
+
+class Divergence(Expr):
+    """
+    Represents unevaluated Divergence.
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, Gradient
+    >>> R = CoordSysCartesian('R')
+    >>> v = R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k
+    >>> Divergence(v, R)
+    Divergence(R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k, R)
+
+    """
+
+    def __new__(cls, vect):
+        obj = Expr.__new__(cls, vect)
+        obj._vect = vect
+        return obj
+
+    def _get_coord_sys_from_expr(self, vect):
+        return vect._sys
+
+    def doit(self):
+        return Del(self._get_coord_sys_from_expr(self._vect)).dot(self._vect).doit()
+
+
+class Curl(Expr):
+    """
+    Represents unevaluated Curl.
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, Curl
+    >>> R = CoordSysCartesian('R')
+    >>> v = R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k
+    >>> Curl(v, R)
+    Curl(R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k, R)
+
+    """
+
+    def __new__(cls, vect):
+        obj = Expr.__new__(cls, vect)
+        obj._vect = vect
+        return obj
+
+    def _get_coord_sys_from_expr(self, vect):
+        return vect._sys
+
+    def doit(self):
+        return Del(self._get_coord_sys_from_expr(self._vect)).cross(self._vect).doit()
+
+
+def gradient(scalar, coord_sys=None):
+    """
+    Returns the vector gradient of a scalar field computed wrt the
+    base scalars of the given coordinate system.
+
+    Parameters
+    ==========
+
+    scalar : SymPy Expr
+        The scalar field to compute the gradient of
+
+    coord_sys : CoordSysCartesian
+        The coordinate system to calculate the gradient in
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, gradient
+    >>> R = CoordSysCartesian('R')
+    >>> s1 = R.x*R.y*R.z
+    >>> gradient(s1)
+    R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k
+    >>> s2 = 5*R.x**2*R.z
+    >>> gradient(s2)
+    10*R.x*R.z*R.i + 5*R.x**2*R.k
+
+    """
+
+    if coord_sys is None:
+        return Gradient(scalar).doit()
+    else:
+        return coord_sys.delop(scalar).doit()
+
+
+def divergence(vect, coord_sys=None):
+    """
+    Returns the divergence of a vector field computed wrt the base
+    scalars of the given coordinate system.
+
+    Parameters
+    ==========
+
+    vect : Vector
+        The vector operand
+
+    coord_sys : CoordSysCartesian
+        The coordinate system to calculate the gradient in
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, divergence
+    >>> R = CoordSysCartesian('R')
+    >>> v1 = R.x*R.y*R.z * (R.i+R.j+R.k)
+    >>> divergence(v1)
+    R.x*R.y + R.x*R.z + R.y*R.z
+    >>> v2 = 2*R.y*R.z*R.j
+    >>> divergence(v2)
+    2*R.z
+
+    """
+
+    if coord_sys is None:
+        return Divergence(vect).doit()
```

Comment:
I would just get the coordinate system like `coord_sys = _get_coord_system_from_expr(vect)`. I am insisting on doing so because I'd like to put more code here in the future.